File: evals/refs/pow2.py
import sys


def main():
    out = []
    for line in sys.stdin:
        line = line.strip()
        if line == "":
            continue
        n = int(line)
        out.append("true" if n > 0 and (n & (n - 1)) == 0 else "false")
    sys.stdout.write("\n".join(out) + ("\n" if out else ""))


if __name__ == "__main__":
    main()

File: evals/refs/pow2.rs
use std::io::{self, Read, Write};

fn main() {
    let mut input = String::new();
    io::stdin().read_to_string(&mut input).unwrap();
    let mut out = String::new();
    for line in input.lines() {
        let line = line.trim();
        if line.is_empty() {
            continue;
        }
        let n: i64 = match line.parse() {
            Ok(v) => v,
            Err(_) => {
                out.push_str("false\n");
                continue;
            }
        };
        if n > 0 && (n & (n - 1)) == 0 {
            out.push_str("true\n");
        } else {
            out.push_str("false\n");
        }
    }
    io::stdout().write_all(out.as_bytes()).unwrap();
}
